src/main/views/segments-layer.js

```diff
@@ -48,6 +48,14 @@ define([
   SegmentsLayer.prototype._registerEventHandlers = function() {
     var self = this;
 
+    this._peaks.on('segments.update', function(segment) {
```

Comment:
We'll need to also update the UI objects if the `startTime` or `endTime` changes. This is a bit more complex than changing the label, as a change to the start/end times could affect whether a segment remains (or becomes) visible in the waveform view.

Reply:
Looking at the event handler for the `segments.add` event, I see this code:

```
this._peaks.on('segments.add', function(segments) {
  var frameOffset = self._view.getFrameOffset();
  var width = self._view.getWidth();

  var frameStartTime = self._view.pixelsToTime(frameOffset);
  var frameEndTime   = self._view.pixelsToTime(frameOffset + width);

  segments.forEach(function(segment) {
    if (segment.isVisible(frameStartTime, frameEndTime)) {
      self._addSegmentGroup(segment);
    }
  });

  self.updateSegments(frameStartTime, frameEndTime);
});
```

It looks like I should just include the same logic in the `segments.updated` event handler?

Reply:
i.e... the value of `isVisible` needs to be reset if necessary, and then `self.updateSegments` should be called with new frameStartTime, frameEndTime?

Reply:
Because `segments.updated` will only receive one segment, we only need to check that particular segment, rather than all of them, so I would avoid calling `updateSegments`. I guess a way to implement would be to remove the segment if it's currently visible, then add it in its new position, maybe something like this:

```javascript
var redraw = false;

var segmentGroup = this._segmentGroups[segment.id];

if (segmentGroup) {
  this._removeSegment(segment);
  redraw = true;
}

var frameOffset = self._view.getFrameOffset();
var width = self._view.getWidth();

var frameStartTime = self._view.pixelsToTime(frameOffset);
var frameEndTime   = self._view.pixelsToTime(frameOffset + width);

if (segment.isVisible(frameStartTime, frameEndTime)) {
  this._addSegmentGroup(segment);
  redraw = true;
}

if (redraw) {
  this._layer.draw();
}
```